application.go

```diff
@@ -144,6 +147,47 @@ type Stats struct {
 	LifeTime map[string]float64 `json:"lifeTime"`
 }
 
+// EnvValue represents either a string environment variable value, or
+// a reference to a Secret defined in the Secrets attribute of the Application
+type EnvValue struct {
+	Value  string
+	Secret string `json:"secret"`
```

Comment:
Instead using a `string` type and subsequently jumping through a few hoops to serialize/serialize secrets properly, how about mapping `Secret` to a struct? Combined with `omitempty` tags, this would save us both the `MarshalJSON` and `UnmarshalJSON` implementations.

(I'm not sure if it makes sense to support empty environment values, but if it does, we can and should pointerize them.)